### PR TITLE
feat(dashboards): Add widget edit and delete overlays

### PR DESCRIPTION
**Pull request by dashed**
Screen Shot 2020-12-04 at 2 04 27 PM

### CODE REVIEW

File: src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx [hunk 1]
```diff
@@ -24,15 +24,23 @@ import {generateFieldOptions} from 'app/views/eventsV2/utils';
 import SelectField from 'app/views/settings/components/forms/selectField';
 import TextField from 'app/views/settings/components/forms/textField';
 
-type Props = ModalRenderProps & {
-  api: Client;
+// forwarded from src/sentry/static/sentry/app/actionCreators/modal.tsx
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:27]: Should we import/export the DashboardWidgetModalOptions and use `Pick` to not duplicate some of the properties?

File: src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx [hunk 2]
```diff
@@ -158,10 +211,12 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
       measurementKeys: [],
     });
 
+    const isUpdatingWidget = typeof onUpdateWidget === 'function' && !!previousWidget;
+
     return (
       <React.Fragment>
         <Header closeButton onHide={closeModal}>
-          {t('Add Widget')}
+          {t(isUpdatingWidget ? 'Edit Widget' : 'Add Widget')}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:219]: I think for the translation parser to work this needs to be:
```suggestion
          {isUpdatingWidget ? t('Edit Widget') : t('Add Widget')}
```

File: src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx [hunk 3]
```diff
@@ -250,7 +308,7 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
                 disabled={state.loading}
                 busy={state.loading}
               >
-                {t('Add Widget')}
+                {t(isUpdatingWidget ? 'Update Widget' : 'Add Widget')}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:311]: I think for the translation parser to work this needs to be:
```suggestion
          {isUpdatingWidget ? t('Update Widget') : t('Add Widget')}
```

File: src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:27]: Would we still have code splitting if we import `DashboardWidgetModalOptions`?

Comment (markstory) [reply src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx:27]: I don't think types should impact code splitting as the output of typescript shouldn't leave imports behind.

**Review by markstory — APPROVED** [review]

**Pull request merged**